Loop/Managers/WatchDataManager.swift

```diff
@@ -334,6 +372,52 @@ extension WatchDataManager {
 
         return items.joined(separator: "\n")
     }
+    
+    private func getSleepStartTime(start: Date, end: Date? = nil, sampleLimit: Int = 30, _ completion: @escaping (_ result: SleepStoreResult<Date>) -> Void) {
+        let predicate = HKQuery.predicateForSamples(withStart: start, end: end, options: [])
+        
+        getSleepStartTime(matching: predicate, sampleLimit: sampleLimit, completion)
+    }
+    
+    private func getSleepStartTime(matching predicate: NSPredicate, sampleLimit: Int, _ completion: @escaping (_ result: SleepStoreResult<Date>) -> Void) {
+        let sleepType = HKObjectType.categoryType(forIdentifier: HKCategoryTypeIdentifier.sleepAnalysis)!
+        // get more-recent values first
+        let sortDescriptor = NSSortDescriptor(key: HKSampleSortIdentifierEndDate, ascending: false)
+        
+        let query = HKSampleQuery(sampleType: sleepType, predicate: predicate, limit: sampleLimit, sortDescriptors: [sortDescriptor]) { (query, samples, error) in
+
+            if let error = error {
+                completion(.failure(error))
+            } else if let samples = samples as? [HKCategorySample] {
+                // find the average hour and minute components from the sleep start times
+                let average = samples.reduce(0, {$0 + $1.startDate.secondsFromMidnight()}) / samples.count
```

Comment:
We'll want a `guard` to check that `!samples.isEmpty` to avoid trapping here if no sleep data is available. SleepStoreError can include a `noSleepDataAvailable` case to represent the failure.